aspects/schema.go

```diff
@@ -444,3 +446,82 @@ func (v *stringSchema) parseConstraints(constraints map[string]json.RawMessage)
 
 	return nil
 }
+
+type intSchema struct {
+	min     *int
+	max     *int
+	choices []int
+}
+
+func (v *intSchema) Validate(raw []byte) error {
+	var num int
+	if err := json.Unmarshal(raw, &num); err != nil {
+		return err
+	}
+
+	if len(v.choices) != 0 {
+		var found bool
+		for _, choice := range v.choices {
+			if num == choice {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return fmt.Errorf(`integer %d is not one of the allowed choices`, num)
+		}
+	}
+
+	if v.min != nil && num < *v.min {
+		return fmt.Errorf(`integer %d is less than allowed minimum %d`, num, *v.min)
+	}
+
+	if v.max != nil && num > *v.max {
+		return fmt.Errorf(`integer %d is greater than allowed maximum %d`, num, *v.max)
+	}
+
+	return nil
+}
+
+func (v *intSchema) parseConstraints(constraints map[string]json.RawMessage) error {
+	if rawChoices, ok := constraints["choices"]; ok {
+		var choices []int
+		err := json.Unmarshal(rawChoices, &choices)
+		if err != nil {
+			return fmt.Errorf(`cannot parse "choices" constraint: %v`, err)
+		}
+
+		if len(choices) == 0 {
+			return fmt.Errorf(`cannot have "choices" constraint with empty list`)
+		}
+
+		v.choices = choices
+	}
+
+	if rawMin, ok := constraints["min"]; ok {
+		if v.choices != nil {
+			return fmt.Errorf(`cannot have "choices" and "min" constraints`)
+		}
+
+		var min int
+		if err := json.Unmarshal(rawMin, &min); err != nil {
+			return fmt.Errorf(`cannot parse "min" constraint: %v`, err)
+		}
+		v.min = &min
+	}
+
+	if rawMax, ok := constraints["max"]; ok {
+		if v.choices != nil {
+			return fmt.Errorf(`cannot have "choices" and "max" constraints`)
+		}
+
+		var max int
+		if err := json.Unmarshal(rawMax, &max); err != nil {
+			return fmt.Errorf(`cannot parse "max" constraint: %v`, err)
+		}
+		v.max = &max
```

Comment:
here we should probably check that if min was also set, min <= max ?

Reply:
Good point, I've added this now. Thanks